packages/dds/matrix/src/matrix.ts

```diff
@@ -308,12 +376,21 @@ export class SharedMatrix<T = any>
 			localSeq,
 			rowsRefSeq,
 			colsRefSeq,
+			referenceSeqNumber: this.runtime.deltaManager.lastSequenceNumber,
 		};
 
 		this.submitLocalMessage(op, metadata);
 		this.pending.setCell(rowHandle, colHandle, localSeq);
 	}
 
+	private protectAgainstReentrancy(callback: () => void) {
```

Comment:
maybe add a comment what it protects against, why it's a bad idea to get reentrant calls